### PR TITLE
Split screenshot iframe into preselection and selection iframes

### LINKED ISSUE #2162: Separate pre-selection iframe from selection iframe
Before you select something we put up an iframe with controls on it.  This iframe should be `position: fixed` (though it currently is not, and we watch for scroll events to move elements around).

After selection we want an iframe that covers the entire document and holds the selection.  This makes scrolling while selecting work.  This is the iframe we have currently.

It would work better if we swap iframes after selection.

### PR DESCRIPTION
**Pull request by dannycoates**
closes #2162

### CODE REVIEW

File: addon/webextension/selector/ui.js [hunk 1]
```diff
@@ -198,83 +201,193 @@ window.ui = (function () { // eslint-disable-line no-unused-vars
...
+    updateElementSize: function (force) {
+      // Note: if someone sizes down the page, then the iframe will keep the
+      // document from naturally shrinking.  We use force to temporarily hide
+      // the element so that we can tell if the document shrinks
+      const visible = this.element.style.display !== "none";
+      if (force && visible) {
+        this.element.style.display = "none";
+      }
+      let height = Math.max(
+        document.documentElement.clientHeight,
```
Comment (ianb) [inline addon/webextension/selector/ui.js]: `updateElementSize` can be simpler for this iframe.  It should always be exactly `window.innerHeight` (and width), and doesn't have to track document changes.  `force` also doesn't matter for this iframe.

File: addon/webextension/selector/ui.js [hunk 2]
```diff
@@ -198,83 +201,193 @@ window.ui = (function () { // eslint-disable-line no-unused-vars
...
-      let scrollX = window.scrollX;
-      let scrollY = window.scrollY;
-      this.movingEl.style.top = scrollY + "px";
-      this.movingEl.style.left = scrollX + "px";
-      this.movingEl.style.width = window.innerWidth + "px";
-      this.movingEl.style.height = window.innerHeight + "px";
+    },
+
+    hide: function () {
+      window.removeEventListener("scroll", this.onScroll, false);
```
Comment (ianb) [inline addon/webextension/selector/ui.js]: Everything related to scroll tracking can be removed.  Neither iframe should require it now.

File: addon/webextension/selector/ui.js [hunk 3]
```diff
@@ -198,83 +201,193 @@ window.ui = (function () { // eslint-disable-line no-unused-vars
...
   };
 
-  WholePageOverlay.onScroll = watchFunction(WholePageOverlay.onScroll.bind(WholePageOverlay));
+  iframePreSelection.onResize = watchFunction(onResize.bind(iframePreSelection));
+
+  let iframe = exports.iframe = {
+    currentIframe: iframePreSelection,
+    display: function (installHandlerOnDocument, standardOverlayCallbacks) {
+      return iframeSelection.display(installHandlerOnDocument)
+        .then(() => iframePreSelection.display(installHandlerOnDocument, standardOverlayCallbacks))
```
Comment (ianb) [inline addon/webextension/selector/ui.js]: `;`

File: addon/webextension/selector/uicontrol.js [hunk 4]
```diff
@@ -636,8 +636,7 @@ window.uicontrol = (function () {
         stateHandlers.resizing.startResize(event, "move");
       } else if (! ui.Box.isControl(target)) {
         mousedownPos = new Pos(event.pageX, event.pageY);
-        mouseupNoAutoselect = true;
-        setState("draggingReady");
+        setState("crosshairs");
```
Comment (ianb) [inline addon/webextension/selector/uicontrol.js]: I suppose we should rename crosshairs as that mode hasn't displayed crosshairs for a long time.  But not really related to this PR.

File: addon/webextension/selector/ui.js (hunk 2 above)
Comment (dannycoates) [inline addon/webextension/selector/ui.js]: The hoverbox remains visible during the scroll without this. I think it looks bad, but if that's not an issue I'll remove it.

Comment (ianb) [inline addon/webextension/selector/ui.js]: Ah, I didn't catch that part.  That makes sense then.

**Pull request merged**